app/javascript/components/patient/close_contacts/CloseContactModal.js

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal, Row, Col, Form } from 'react-bootstrap';
+
+import { phoneSchemaValidator } from '../../../utils/Patient';
+
+import _ from 'lodash';
+import moment from 'moment';
+import * as yup from 'yup';
+
+import DateInput from '../../util/DateInput';
+import ReactTooltip from 'react-tooltip';
+import PhoneInput from '../../util/PhoneInput';
+import InfoTooltip from '../../util/InfoTooltip';
+
+const MAX_NOTES_LENGTH = 2000;
+
+class CloseContactModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      first_name: props.currentCloseContact.first_name,
+      last_name: props.currentCloseContact.last_name,
+      primary_telephone: props.currentCloseContact.primary_telephone,
+      email: props.currentCloseContact.email,
+      last_date_of_exposure: props.currentCloseContact.last_date_of_exposure,
+      assigned_user: props.currentCloseContact.assigned_user,
+      contact_attempts: props.currentCloseContact.contact_attempts,
+      notes: props.currentCloseContact.notes || '',
+      errors: {},
+      isValid:
+        (props.currentCloseContact.first_name || props.currentCloseContact.last_name) &&
+        (props.currentCloseContact.primary_telephone || props.currentCloseContact.email),
+    };
+  }
+
+  validateAndSubmit = () => {
+    schema
+      .validate({ ...this.state }, { abortEarly: false })
+      .then(() => {
+        this.props.onSave(this.state);
+      })
+      .catch(err => {
+        // Validation errors, update state to display to user
+        if (err && err.inner) {
+          let issues = {};
+          for (const issue of err.inner) {
+            issues[issue['path']] = issue['errors'];
+          }
+          this.setState({ errors: issues });
+        }
+      });
+  };
+
+  handleDateChange = event => this.setState({ last_date_of_exposure: event });
+
+  handleChange = event => {
+    let stateKey = _.startsWith(event.target.id, 'cc_') ? _.trim(event.target.id, 'cc_') : event.target.id;
+    if (event?.target?.value && typeof event.target.value === 'string' && event.target.value.match(/^\s*$/) !== null) {
+      // Empty spaces are allowed to be typed (for example, a first name may be 'Mary Beth')
+      // But empty starting first spaces should not be allowed
+      event.target.value = '';
+    }
+    let value;
+    if (stateKey === 'assigned_user') {
```

Comment:
This method seems excessively complex with the string manipulation on IDs above and then the conditionals below. How would you feel about splitting up this `handleChange` method into smaller ones for each field, like we do in the [`VaccineModal`](https://github.com/SaraAlert/SaraAlert/blob/1.35.0/app/javascript/components/patient/vaccines/VaccineModal.js#L34-L54)?

Reply:
Um I could see an argument either way. But you are definitely correct that it's difficult to read this. I'm gonna break it up like you suggest for readability concerns.